src/xrGame/xrServer.cpp

```diff
@@ -866,7 +866,7 @@ void xrServer::Server_Client_Check(IClient* CL)
         return;
     };
 
-    if (CL->process_id == GetCurrentProcessId())
+    if (CL->process_id == static_cast<u32>(GetCurrentProcessId()))
```

Comment:
On Windows `process_id` variable has `DWORD` type and `GetCurrentProcessId()` returns `DWORD` too.
On Linux this variable has type `pid_t` and GetCurrentProcessId() is a macro to `getpid` function that returns `pid_t`.

Why does it throw a warning here??

Reply:
As with all the changes in this commit, the warning is for comparing a signed int with an unsigned

Reply:
Try this: will it fix the warning?
```suggestion
    if (CL->process_id == static_cast<xrpid_t>(GetCurrentProcessId()))
```

Reply:
The warning persist, xrpid_t is consider an int.
How about converting process_id to int for this comparison? unsigned to signed should be safer